Wrappers/Python/test/test_algorithms.py

```diff
@@ -1131,7 +1147,210 @@ def test_SIRT_with_TV_warm_start(self):
+
+        with self.assertRaises(ValueError):
+            spdhg = SPDHG(f=self.F, g=self.G, operator=self.A, _norms=[1]*len(self.A), prob=[1/(self.subsets-1)]*(
+                self.subsets-1)+[0], sampler=Sampler.random_with_replacement(10, list(np.arange(1, 11)/55.)))
+            
+            
+        with self.assertRaises(ValueError):
+            spdhg = SPDHG(f=self.F, g=self.G, operator=self.A, _norms=[1]*len(self.A), prob=[1/(self.subsets-1)]*(
+                self.subsets-1)+[0], prob_weights= [1/(self.subsets-1)]*(self.subsets-1)+[0])
+
+        with self.assertRaises(ValueError):
+            spdhg = SPDHG(f=self.F, g=self.G, operator=self.A, sfdsdf=3,  _norms=[
+                          1]*len(self.A), sampler=Sampler.random_with_replacement(10, list(np.arange(1, 11)/55.)))
+
+
+    def test_spdhg_set_norms(self):
+
+        self.A2.set_norms([1]*len(self.A2))
+        spdhg = SPDHG(f=self.F, g=self.G, operator=self.A2)
+        self.assertListEqual(spdhg._norms, [1]*len(self.A2))
+
+    def test_spdhg_check_convergence(self):
+        spdhg = SPDHG(f=self.F, g=self.G, operator=self.A)
+
+        self.assertTrue(spdhg.check_convergence())
+
+        gamma = 3.7
+        rho = 0.9
+        spdhg.set_step_sizes_from_ratio(gamma, rho)
+        self.assertTrue(spdhg.check_convergence())
+
+        gamma = 3.7
+        rho = 100
+        spdhg.set_step_sizes_from_ratio(gamma, rho)
+        self.assertFalse(spdhg.check_convergence())
+
+        spdhg.set_step_sizes(sigma=[1]*self.subsets, tau=100)
+        self.assertFalse(spdhg.check_convergence())
+
+        spdhg.set_step_sizes(sigma=[1]*self.subsets, tau=None)
+        self.assertTrue(spdhg.check_convergence())
+
+        spdhg.set_step_sizes(sigma=None, tau=100)
+        self.assertTrue(spdhg.check_convergence())
+
+    def test_SPDHG_num_subsets_1(self): 
+        data = dataexample.SIMPLE_PHANTOM_2D.get(size=(10, 10))
+
+        subsets = 1
+
+        ig = data.geometry
+        ig.voxel_size_x = 0.1
+        ig.voxel_size_y = 0.1
+
+        detectors = ig.shape[0]
+        angles = np.linspace(0, np.pi, 90)
+        ag = AcquisitionGeometry.create_Parallel2D().set_angles(
+            angles, angle_unit='radian').set_panel(detectors, 0.1)
+        # Select device
+        dev = 'cpu'
+
+        Aop = ProjectionOperator(ig, ag, dev)
+
+        sin = Aop.direct(data)
+        partitioned_data = sin.partition(subsets, 'sequential')
+        A = BlockOperator(
+            *[IdentityOperator(partitioned_data[i].geometry) for i in range(subsets)])
+       
+        # block function
+        F = BlockFunction(*[L2NormSquared(b=partitioned_data[i])
+                                 for i in range(subsets)])
+        
+        F_phdhg=L2NormSquared(b=partitioned_data[0])
+        A_pdhg = IdentityOperator(partitioned_data[0].geometry) 
+                            
+        alpha = 0.025
+        G = alpha * FGP_TV()
```

Comment:
Can we test this with a native CIL function to remove the tests additional dependency